"Migration guide" for helping users make published RAML specs compliant with RAML 0.8 and 1.0 (#6)
mulesoft/docs-design-center: modules/ROOT/pages/design-modify-raml-specs-conform.adoc

= Modify Published RAML API Specifications to Conform Completely to RAML 0.8 or 1.0

[[bookmark-a,Back to the top]]

> As of November 1, 2018, the API editor Design Center is now more precise in ensuring that the RAML in your API specifications is valid. However, it is possible that one or more of the APIs that you used previous versions of the API editor to create and publish to Anypoint Exchange includes invalid RAML that those previous versions did not flag.

Reviewer: I'm thinking if we don't need to write "Anypoint Design Center", we can also omit the "Anypoint" from Exchange.

```suggestion
As of November 1, 2018, the API editor in Design Center is now more precise in ensuring that the RAML in your API specifications is valid. However, it is possible that one or more of the APIs that you used previous versions of the API editor to create and publish to Exchange includes invalid RAML that those previous versions did not flag.
```

> Check your published APIs by opening them in the API editor in Design Center. You can use the guide on this page to correct a number of errors that MuleSoft has found while testing the current API editor on a large number of RAML API specifications. If you see errors that this guide does not include, refer to the specifications for RAML 0.8 and 1.0, which you can read on https://raml.org/[raml.org].

Reviewer: We started short-handing "RAML API specification" for just "RAML" in the next paragraph. Should be consistent here.

```suggestion
Check your published APIs by opening them in the API editor in Design Center. You can use the guide on this page to correct a number of errors that MuleSoft has found while testing the current API editor on a large number of RAML. If you see errors that this guide does not include, refer to the specifications for RAML 0.8 and 1.0, which you can read on https://raml.org/[raml.org].
```

Author: "a large number of RAML" doesn't make sense here.

> You have until November 1, 2019 to correct any invalid RAML that your published API specifications might include. After that date, you will be unable to republish and redeploy APIs in which there is invalid RAML.

Reviewer: @chiragnarangmulesoft and I talked about this date being more clear about this not being a typo.

```suggestion
You have until October 31, 2019 to correct any invalid RAML that your published API specifications might include. After that date, you will be unable to republish and redeploy APIs in which there is invalid RAML.
```

* <<bookmark-b,Common Errors in Conforming to RAML 0.8>>
* <<bookmark-c,Common Errors in Conforming to RAML 1.0>>

[[bookmark-b,Common Errors in Conforming to RAML 0.8]]
== Common Errors in Conforming to RAML 0.8

If you open in the API editor a published API specification that is written in RAML 0.8 and you see error indicators, this list might help you to resolve the errors. If you see errors that this list does not include, refer to the specification for RAML 0.8, which you can read on https://raml.org/[raml.org].

=== Common Error 1 (RAML 0.8)
// APIMF-824

*_Not using a data type of string for the example of a Boolean enum_*

Whether the values for a Boolean enum are both strings or Boolean values, the example of the enum must be a string. To illustrate the error, this snippet shows an example using the wrong data type.

----
#%RAML 0.8
title: ExampleRAML
/nob:
  head:
    responses:
      204:
        headers:
          X-NOB-Exists:
            enum: [true, false]
            example: true
----

> RAML 0.8 supports enum values as strings only, not as Boolean values. Also, in the example RAML above, no data type for the enum is declared. In RAML 0.8, when a data type is not declared, it is assumed to be `string`.

Reviewer: ```suggestion
RAML 0.8 supports enum values as strings only, not as boolean values. Also, in the example RAML above, no data type for the enum is declared. In RAML 0.8, when a data type is not declared, it is assumed to be `string`.
```

> Therefore, the values in the enum in the incorrect example above must be in quotation marks. Because the data type of an example must be the same as the data type of the element that it is an example of, the value in the example must be in quotation marks, too.

Reviewer: I'm having the hardest time parsing this sentence. Maybe this is slightly more clear?

```suggestion
Therefore, the values in the enum in the incorrect example above must be in quotation marks. Because the data type of an example must match the data type of the element that it is an example of, the value in the example must be in quotation marks, too.
```

Author: Fixed.
